### PR TITLE
Refactor/more redux reducers

### PR DESCRIPTION
**Pull request by JiaqiWang18**
## Description
Refactor more redux reducers

## Progress
Current Progress: 11 completed, 8 left to be completed in another PR

### Refactored the following reducer into redux toolkit slice
* Preferences reducer
* Registrar reducer
* Save Calendar Modal reducer
* School reducer
* Search Results reducer
* Semester reducer
* Terms of Service Banner reducer
* Terms of Service Modal reducer
* Ui reducer

### CODE REVIEW

**Comment by JiaqiWang18** [comment]
seems like GitHub actions is down

**Comment by JJamesWWang** [comment]
This is gonna be a massive PR to review if you finish it al; can you mark this one as ready and create another one?

**Comment by JJamesWWang** [comment]
Could you also note down which reducers were refactored in the description?
(actually, just check if I messed any)

File: static/js/redux/actions/timetable_actions.jsx [hunk 1]
```diff
@@ -146,7 +147,7 @@ export const lockTimetable = (timetable) => (dispatch, getState) => {
   const state = getState();
 
   if (timetable.has_conflict) {
-    dispatch({ type: ActionTypes.TURN_CONFLICTS_ON });
+    dispatch(preferencesActions.turnConflictsOn());
```
Comment (JJamesWWang) [inline static/js/redux/actions/timetable_actions.jsx:150]: Thinking of getting rid of the preferences modal soon, what do you think?

**Review by JJamesWWang — APPROVED** [review]
Looks good, check if the description covers everything and comment if it's right

**Comment by JiaqiWang18** [comment]
> Could you also note down which reducers were refactored in the description? (actually, just check if I messed any)

Thanks lol, I think that's all

File: static/js/redux/actions/timetable_actions.jsx (hunk 1 above)
Comment (JiaqiWang18) [reply static/js/redux/actions/timetable_actions.jsx:150]: Yes, the preference modal is kinda buggy, I think we should remove

**Comment by JiaqiWang18** [comment]
> Looks good, check if the description covers everything and comment if it's right

It's good

File: static/js/redux/actions/timetable_actions.jsx (hunk 1 above)
Comment (JJamesWWang) [reply static/js/redux/actions/timetable_actions.jsx:150]: Okay, let's do that after redux reducers are refactored

**Pull request merged**